### ISSUE TITLE
Support returning a Hash

### ISSUE THREAD

**Issue by amayer171** [procore/blueprinter#73]
It would be helpful in some cases to return a hash.

This is already possible with the `prepare` method. https://github.com/procore/blueprinter/blob/master/lib/blueprinter/base.rb#L141 but it is listed as private and does not support the same interface as render. 

Suggested method signature could be `render_as_hash(object, options = {})`

**Comment by philipqnguyen** [comment]
I'm curious to know in what use case you are running into that returning a hash would be needed?

**Comment by ojab** [comment]
I'm using server-side data normalization (and not using blueprinter [yet]) and all models are returned in a `{ [model_name]: [serialized_array] }` hash, so something like 
```ruby
render json: { first_models: FirstModelBlueprint.render_as_hash(first_models),
               second_models: SecondModelBlueprint.render_as_hash(second_models) }
```
will be useful.
What is the proper way to do it right now?

**Comment by philipqnguyen** [comment]
@ojab Totally makes sense. Right now there is no proper way of doing it with Blueprinter. I think what @amayer171 suggested about creating a new `render_as_hash` that utilizes the `prepare` method makes sense. Would any of you want to take a stab at a PR for it?

**Comment by ojab** [comment]
I'd like yo take a stab, but `render_as_hash` name looks quite ugly (and long) to me tbh. 
Since https://github.com/procore/blueprinter/issues/67 is not closed yet — maybe it should be `publish`/`prepublish`? Or just plain and understandable `to_json`/`as_json`?

**Comment by amayer171** [comment]
My use case is that I'd like to start using blueprinter for nested objects in my existing serializers. These existing serializers expect a hash rather than a string because they provide their own stringify implementation so that they can stringify in the last step. Otherwise the blueprinter result will be a string literal with hash syntax escaped. 

- render_as_hash is a little ugly, but it certainly does what it says. 
- publish / prepublish seems to abstract.
- as_json is the existing method in AMS: http://api.rubyonrails.org/classes/ActiveModel/Serializers/JSON.html#method-i-as_json. Personally I don't love it since JSON has the connotation of being a string so that can cause some confusion.

**Comment by allanrsmith** [comment]
+1 `render_as_hash`

**Comment by dlcarter** [comment]
To mangle a Churchill quote, I think `render_as_hash` is the worst option, except for all the others. So I'd +1 that as well

**Comment by philipqnguyen** [comment]
#73 merged and released in 0.4.0
